osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/configuration/service/SetConfigurationObjectServiceTest.java

```diff
@@ -0,0 +1,101 @@
+package org.opensmartgridplatform.adapter.protocol.dlms.domain.commands.configuration.service;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.when;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Optional;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.openmuc.jdlms.DlmsConnection;
+import org.openmuc.jdlms.SetParameter;
+import org.openmuc.jdlms.datatypes.DataObject;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.commands.utils.DlmsHelper;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.entities.Protocol;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.factories.DlmsConnectionManager;
+import org.opensmartgridplatform.adapter.protocol.dlms.exceptions.ConnectionException;
+import org.opensmartgridplatform.adapter.protocol.dlms.exceptions.ProtocolAdapterException;
+import org.opensmartgridplatform.adapter.protocol.dlms.infra.messaging.DlmsMessageListener;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.ConfigurationFlagDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.ConfigurationFlagTypeDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.ConfigurationFlagsDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.ConfigurationObjectDto;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SetConfigurationObjectServiceTest {
+
+    private SetConfigurationObjectService instance;
+
+    @Mock
+    private DlmsHelper dlmsHelper;
+    @Mock
+    private DlmsConnectionManager conn;
+    @Mock
+    private DlmsMessageListener dlmsMessageListener;
+    @Mock
+    private DlmsConnection dlmsConnection;
+    @Mock
+    private ConfigurationObjectDto configurationToSet;
+    @Mock
+    private ConfigurationObjectDto configurationOnDevice;
+
+    @Before
+    public void setUp() {
+        when(this.conn.getDlmsMessageListener()).thenReturn(this.dlmsMessageListener);
+        when(this.conn.getConnection()).thenReturn(this.dlmsConnection);
+
+        this.instance = new SetConfigurationObjectService(this.dlmsHelper) {
```

Comment:
why create an instance with a dummy impl while this is the class under test?

Reply:
Because the class under test is abstract.